### PR TITLE
Seperate docker-compose.yml file

### PR DESCRIPTION
**Pull request by khoahd7621**
Hello everyone,

I just break the old docker-compose.yml file into multiple files such as:
docker-compose.yml for core services
docker-compose.search.yml for elastic search-related services
docker-compose.o11y.yml for observability-related services

Please help me review this PR, thanks.

### CODE REVIEW

**Comment by thiennn** [comment]
Please also update the README on how to run the application.

File: docker-compose.o11y.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+version: "3.9"
+
+services:
+  loki:
```
Comment (thiennn) [inline docker-compose.o11y.yml:4]: the services (product, order,....) will send log and trace data to loki and tempo. Please help me verify that they can work fine even the loki, tempo is not up and running

File: docker-compose.search.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    environment:
+      SERVER_SERVLET_CONTEXT_PATH: /search
+      YAS_PUBLIC_URL: http://api.yas.local/search
+      YAS_SERVICES_PRODUCT: http://product/product
+      SERVER_PORT: "80"
+    depends_on:
+      - kafka
+    networks:
+      - yas-network
+  zookeeper:
```
Comment (thiennn) [inline docker-compose.search.yml:16]: zookeeper, kafka, kafka connect, akhq not just be used to feed data for elastic search. I think we should put them to the core services (docker-compose.yml)

File: docker-compose.o11y.yml (hunk 1 above)
Comment (khoahd7621) [reply docker-compose.o11y.yml:4]: These exceptions will disappear after I run docker compose for observation.

File: docker-compose.search.yml (hunk 2 above)
Comment (khoahd7621) [reply docker-compose.search.yml:16]: Currently, kafka-connect depends on elasticsearch. Let me try kafka-connect still works without depending on elasticsearch.

Comment (khoahd7621) [reply docker-compose.search.yml]: I moved them to the core services.

**Review by LEHOANGGLAM — APPROVED** [review]

**Comment by thiennn** [comment]
I think we should move loki and tempo to the core services. Also put a note to warn people that they will need at least 16GB RAM available to run all the services. If you don't have that amount of RAM you can run only core services.

**Comment by khoahd7621** [comment]
> I think we should move loki and tempo to the core services. Also put a note to warn people that they will need at least 16GB RAM available to run all the services. If you don't have that amount of RAM you can run only core services.

Okay.

**Review by thiennn — APPROVED** [review]

**Pull request merged**